lumicks/pylake/channel.py

```diff
@@ -21,14 +22,54 @@ class Slice:
         Plot labels: "x", "y", "title".
     calibration: ForceCalibration
     """
-    def __init__(self, data_source, labels=None, calibration=None):
+    def __init__(self, data_source, labels=None, calibration=None, unit=None):
         self._src = data_source
         self.labels = labels or {}
+        self.unit = unit
         self._calibration = calibration
 
     def __len__(self):
         return len(self._src)
 
+    def _unpack_other(self, other):
+        if isinstance(other, self.__class__):
+            return other._src.data
+        else:
+            return other
+
+    def compare_unit(self, other, operation):
+        other_unit = getattr(other, "unit", None)
+        if other_unit:
+            return determine_unit(self.unit, other_unit, operation)
+
+    def __add__(self, other):
+        unit = self.compare_unit(other, 'add')
+        return Slice(self._src._with_data_source(self._src.data + self._unpack_other(other)), unit=unit)
```

Comment:
My python is old and crappy so I may be wrong.

Aren't we constructing `Slice` twice here? one inside `_with_data_source` and another explicitly?  
If so we can probably drop the explicit one.

Reply:
Nah, the function _with_data_source generates a new instance of the underlying datatype (Continuous, TimeSeries or TimeTags), not slice (note the _src).